rmw_connextdds_common/src/common/rmw_context.cpp

```diff
@@ -184,6 +184,25 @@ rmw_context_impl_t::initialize_node(
 
   this->qos_library = qos_library;
 
+  /* Lookup name of custom QoS library */
+  const char * do_not_override_publish_mode_env = nullptr;
+  lookup_rc = rcutils_get_env(
+    RMW_CONNEXT_ENV_DO_NOT_OVERRIDE_PUBLISH_MODE, &do_not_override_publish_mode_env);
+
+  if (nullptr != lookup_rc || nullptr == qos_library) {
```

Comment:
I noticed after merging that this check should be for `do_not_override_publish_mode_env` and not `qos_library`. I can fix this in another commit, since I'm working on reorganizing the participant factory's initialization/finalization.

Reply:
Ups, wrong copy pasting

Reply:
I opened https://github.com/rticommunity/rmw_connextdds/pull/6 with a direct fix, but feel free to close that PR and commit the fix elsewhere is that easier.